core/src/smartcontracts/wasm.rs

```diff
@@ -234,12 +275,17 @@ impl<'wrld, W: WorldTrait> Runtime<'wrld, W> {
     }
 
     fn create_smart_contract(
-        &self,
+        &mut self,
         store: &mut Store<State<'wrld, W>>,
+        iroha_instance: Instance,
         bytes: impl AsRef<[u8]>,
-    ) -> Result<wasmtime::Instance, anyhow::Error> {
-        let module = Module::new(&self.engine, bytes)?;
-        self.linker.instantiate(store, &module)
+    ) -> Result<Instance, anyhow::Error> {
```

Comment:
This method links and instantiates the smartcontract module (as `bytes`) and the memory assistant module (as `iroha_instance`), right?

Reply:
this instantiates only the smart contract, `iroha_instance` was instantiated earlier and is used here to import into smart contract all the exported items from `iroha_instance`

Reply:
So effectively we have a `memcpy` of the functions. To me this seems the best that can be done under the circumstances.